src/Timer_TEST.cc

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <gtest/gtest.h>
+
+#include <thread>
+
+#include <ignition/common/Timer.hh>
+
+/////////////////////////////////////////////////
+TEST(Timer_TEST, Sequence)
+{
+  ignition::common::Timer t;
+  EXPECT_FALSE(t.Running());
+  EXPECT_DOUBLE_EQ(0.0, t.ElapsedTime().count());
+
+  // Call stop on a timer that hasn't been started
+  t.Stop();
+  EXPECT_FALSE(t.Running());
+
+  // Start the timer
+  t.Start();
+  EXPECT_TRUE(t.Running());
+  EXPECT_LT(0.0, t.ElapsedTime().count());
+
+  t.Stop();
+  EXPECT_FALSE(t.Running());
+  EXPECT_LT(0.0, t.ElapsedTime().count());
+}
+
+/////////////////////////////////////////////////
+TEST(Timer_TEST, Elapsed)
+{
+  ignition::common::Timer t;
+  t.Start();
+  std::this_thread::sleep_for(std::chrono::milliseconds(50));
+  t.Stop();
+
+  // Loose margin just to make sure it is close without being flaky
+  EXPECT_NEAR(0.05, t.ElapsedTime().count(), 0.01);
```

Comment:
https://github.com/ignitionrobotics/ign-common/runs/4942752187?check_suite_focus=true

Reply:
Well, I thought that would be wide enough for any flakiness.  Maybe need to eliminate any time-based if the macOS actions are that bad.

Reply:
Changed to just make sure elapsed time is more than 0.